code/controllers/subsystem/ships.dm

```diff
@@ -263,69 +300,103 @@ var/global/list/ftl_weapons_consoles = list()
-				S.called_for_help = 1
-				spawn(0)
-					distress_call(SSstarmap.current_system)
-	if(S.planet != SSstarmap.current_planet && prob(1) && !S.target && !check_hostilities(S.faction,"ship"))
-		broadcast_message("<span class=warning>Enemy ship ([S.name]) at [S.planet] powering up FTL drive for interplanetary jump.</span>",alert_sound)
-		S.is_jumping = 1
-		S.target = SSstarmap.current_planet
+	S.mission_ai.fire(S)
+	S.operations_ai.fire(S)
+	S.combat_ai.fire(S)
 
 
 
 /datum/subsystem/ship/proc/process_ftl(var/datum/starship/S)
 	if(!S.is_jumping)
 		return
+
 	S.jump_progress += round(S.evasion_chance / initial(S.evasion_chance))
 	if((S.jump_progress >= S.jump_time) && !S.target)
-		broadcast_message("<span class=notice>Enemy ship ([S.name]) successfully charged FTL drive. Enemy ship has left the system.</span>",notice_sound)
-		qdel(S)
+		broadcast_message("<span class=notice>[faction2prefix(S)] ship ([S.name]) successfully charged FTL drive. [faction2prefix(S)] ship has left the system. Destination vector: ([S.ftl_vector.name])</span>",notice_sound,S)
+		S.is_jumping = 0
+		remove_system(S,S.system)
+		S.jump_progress = 0
+		spawn transit_system(S)
 	if((S.jump_progress >= S.jump_time / 2) && S.target)
-		broadcast_message("<span class=notice>Enemy ship ([S.name]) sucessfully jumped to [S.target].</span>",notice_sound)
+		broadcast_message("<span class=notice>[faction2prefix(S)] ship ([S.name]) sucessfully jumped to [S.target].</span>",notice_sound,S)
 		S.planet = S.target
 		S.is_jumping = 0
+		S.jump_progress = 0
+
+/datum/subsystem/ship/proc/distress_call(var/datum/starship/caller,var/player_distress,var/datum/starship/target)
+	if(caller.system.capital_planet)
+		return //if the person calling for help is in the capital.. well screw you. Even if you're in the enemy capital you're on your own
+
+	var/chance
+	if(player_distress)
+		chance = max(5,(heat_level * 5 + (caller.system.danger_level - 1) * 10) - 30)
+		last_known_player_system = caller.system
 
-/datum/subsystem/ship/proc/distress_call(var/datum/star_system/system)
-	sleep(100)
-	if(system != SSstarmap.current_system)
-		return
-	var/num_ships = 0
-	if(prob(1))
-		priority_announce("Large enemy fleet movements detected on long range sensors closing on your position. Recommended course of action: Get the fuck out of there.")
-		num_ships = rand(8,20)
 	else
-		num_ships = rand(1,4)
-	sleep(1200)
-	if(system != SSstarmap.current_system)
-		return
-	SSstarmap.generate_npc_ships(num_ships)
-	broadcast_message("<span class=warning>Warning: [num_ships] enemy contacts detected jumping into system.</span>",alert_sound)
+		chance = max(5,((caller.system.danger_level - 1) * 10) - 30)
+		target.last_known_system = caller.system
+
+	if(prob(chance))
+		var/datum/star_faction/faction = cname2faction(caller.faction)
+		var/list/possible_ships = list()
+
+		for(var/datum/starship/S in faction.ships)
+			if(S.mission_ai.cname = "MISSION_PATROL") //get roaming fleets to go first
```

Comment:
I know, I found this in the travis error, but DM doesn't find this anymore because Lummox is retarded and made this an actual feature.